From b23f479913787c2dfac7a811780fd0f957f0fcb3 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:41 -0400
Subject: [ethernet] i40e/i40evf: clear aq bah-bal on shutdown

Message-id: <1409658042-22689-221-git-send-email-sassmann@redhat.com>
Patchwork-id: 90408
O-Subject: [RHEL7.1 PATCH 220/281] i40e/i40evf: clear aq bah-bal on shutdown
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

Clear the AQ BAH and BAL registers on a clean shutdown to help make sure
all is tidy when the driver is done.

Change-ID: I393e92680247daa52a8e00bab183213672d73578
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 4346940b961606d0628ffa25780797b2e611e9b4)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_adminq.c b/drivers/net/ethernet/intel/i40e/i40e_adminq.c
index 87f1d8b..a8244a5 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_adminq.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_adminq.c
@@ -489,6 +489,8 @@ static i40e_status i40e_shutdown_asq(struct i40e_hw *hw)
  wr32(hw, hw->aq.asq.head, 0);
  wr32(hw, hw->aq.asq.tail, 0);
  wr32(hw, hw->aq.asq.len, 0);
+ wr32(hw, hw->aq.asq.bal, 0);
+ wr32(hw, hw->aq.asq.bah, 0);
 
  /* make sure lock is available */
  mutex_lock(&hw->aq.asq_mutex);
@@ -520,6 +522,8 @@ static i40e_status i40e_shutdown_arq(struct i40e_hw *hw)
  wr32(hw, hw->aq.arq.head, 0);
  wr32(hw, hw->aq.arq.tail, 0);
  wr32(hw, hw->aq.arq.len, 0);
+ wr32(hw, hw->aq.arq.bal, 0);
+ wr32(hw, hw->aq.arq.bah, 0);
 
  /* make sure lock is available */
  mutex_lock(&hw->aq.arq_mutex);
-- 
1.7.1